### PR TITLE
Improvement

### PR DESCRIPTION
**Pull request by vigneshwarn**
1. Added status code 429 Exception
2. Added created/updated time in ZonedDateTime format

### CODE REVIEW

**Comment by shivam1608** [comment]
👋 thank you for showing interest in this project.
but do we really need `ZonedDateTime` ? i don't think so because the developer using the library can easily parse that thing i didn't find it useful 
however your update `1.Added status code 429 Exception` is reasonable so you have to remove the `ZonedDateTime` so i can merge the request also kindly revert the changes in `pom.xml`  regarding version because it gets changed in major updates 
if you wish you can update the documentation too.

**Comment by vigneshwarn** [comment]
@shivam1608  I have seen most of the developers were using ZonedDateTime. Which will be very useful for comparison/verification/Time-related wait etc......, I was using your Lib for a very long back thought this thing is missing, what do u feel. Also the String createdAt & String updatedAt are still available.  After ur confirmation i can will update the PR

File: src/main/java/me/shivzee/util/JMailBuilder.java [hunk 1]
```diff
@@ -87,6 +87,8 @@ public static JMailTM createAndLogin(String email , String password) throws Logi
 
             }else if(response.getResponseCode() == 422){
                 throw new LoginException("Account Already Exists! Error 422");
+            }else if (response.getResponseCode() == 429){
+                throw new LoginException("Too Many account create request");
```
Comment (shivam1608) [inline src/main/java/me/shivzee/util/JMailBuilder.java]: to be specific its `Too many requests! Error 429 Ratelimited`

File: src/main/java/me/shivzee/util/Message.java [hunk 2]
```diff
@@ -207,6 +212,31 @@ public String getCreatedAt() {
...
 
+    /**
+     * Get the Message Received Date/Time in ZonedDateTime format
+     * @return the date at which the message was sent/created/received
+     */
+    public ZonedDateTime getCreatedDateTime() {
+        return createdDateTime;
+    }
+
+    public void setCreatedDateTime(ZonedDateTime createdDateTime) {
```
Comment (shivam1608) [inline src/main/java/me/shivzee/util/Message.java:223]: there is no need of a setter over here this class just binds the data and it should be immutable 
remove this function `setCreatedDateTime(ZonedDateTime)`

File: src/main/java/me/shivzee/util/Message.java [hunk 3]
```diff
@@ -207,6 +212,31 @@ public String getCreatedAt() {
...
+    /**
+     * Get the Message Update Date/Time in  ZonedDateTime format
+     * @return the date on which the message was updated (markAsRead fires the update event)
+     * @see me.shivzee.callbacks.EventListener
+     */
+    public ZonedDateTime getUpdatedDateTime() {
+        return updatedDateTime;
+    }
+
+    public void setUpdatedDateTime(ZonedDateTime updatedDateTime) {
```
Comment (shivam1608) [inline src/main/java/me/shivzee/util/Message.java:236]: as mentioned above reason for `setCreatedDateTime`
remove this function `setUpdatedDateTime(ZonedDateTime)`

File: src/main/java/me/shivzee/util/Utility.java [hunk 4]
```diff
@@ -24,5 +28,22 @@ public static String createRandomString(int length){
...
+            try {
+                dateTime = dateTime.replace("AM", "am");
+                dateTime = dateTime.replace("PM", "pm");
+                LocalDateTime objLocalDateTime = LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern(pattern));
+                return objLocalDateTime.atZone(ZoneId.systemDefault());
+            }catch(Throwable t) {
+            }
+        }
+        return null;
+    }
```
Comment (shivam1608) [inline src/main/java/me/shivzee/util/Utility.java]: badly written function 
the `parseDateTimeToDefaultTimeZone` can be written in short as `parseToDefaultTimeZone` 
and the function can be simplified as
```java
public static ZonedDateTime parseToDefaultTimeZone(String dateTime, String pattern) {
        ZonedDateTime time = null;
        try {
            time = LocalDateTime.parse(dateTime, DateTimeFormatter
            .ofPattern(pattern))
            .atZone(ZoneId.systemDefault());
        }catch(Exception ex) {
            //TODO -> Add the LOG.Error();
        }
        return time;
    }
```
there is no need for `AM`,`PM` its a UTC timestamp which mail.tm api follows
also add a little bit info in the documentation comment above for better understanding of what this function does

```suggestion
        ZonedDateTime time = null;
        try {
            time = LocalDateTime.parse(dateTime, DateTimeFormatter
            .ofPattern(pattern))
            .atZone(ZoneId.systemDefault());
        }catch(Exception ex) {
            //TODO -> Add the LOG.Error();
        }
        return time;
        }

```

File: src/main/java/me/shivzee/util/Message.java (hunk 2 above)
Comment (shivam1608) [inline src/main/java/me/shivzee/util/Message.java]: there is no need of a setter over here this class just binds the data and it should be immutable
remove this function `setCreatedDateTime(ZonedDateTime)`

File: src/main/java/me/shivzee/util/Message.java (hunk 3 above)
Comment (shivam1608) [inline src/main/java/me/shivzee/util/Message.java]: as mentioned above reason for `setCreatedDateTime`
remove this function `setUpdatedDateTime(ZonedDateTime)`

**Review by shivam1608 — CHANGES_REQUESTED** [review]
after thinking the use case of `ZonedDateTime` i think we should add it to the library so kindly fix these issue to continue merge

File: src/main/java/me/shivzee/util/JMailBuilder.java (hunk 1 above)
Comment (vigneshwarn) [reply src/main/java/me/shivzee/util/JMailBuilder.java]: Added

File: src/main/java/me/shivzee/util/Message.java (hunk 3 above)
Comment (vigneshwarn) [reply src/main/java/me/shivzee/util/Message.java]: Removed

**Review by shivam1608 — APPROVED** [review]

**Comment by shivam1608** [comment]
nice work 👍will be usable in the next release
till then you can use `main-SNAPSHOT` as version if you want to use before release

**Pull request merged**